nixos/modules/services/hardware/dell-bios-fan-control.nix

```diff
@@ -0,0 +1,47 @@
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}:
+let
+  cfg = config.services.hardware.dell-bios-fan-control;
+in
+{
+  meta.maintainers = with lib.maintainers; [ rickyelopez ];
+
+  options.services.hardware.dell-bios-fan-control = {
+    enable = lib.mkEnableOption "One-shot service to disable dell bios fan control on startup";
+    package = lib.mkPackageOption pkgs "dell-bios-fan-control" { };
+  };
+
+  config = lib.mkIf cfg.enable {
+    environment.systemPackages = [ cfg.package ];
+    # see ref in aur: https://aur.archlinux.org/cgit/aur.git/tree/dell-bios-fan-control.service?h=dell-bios-fan-control-git
+    systemd.services.dell-bios-fan-control = {
+      description = "Disables BIOS control of fans at boot.";
+      wants = [ "dell-bios-fan-control-resume.service" ];
+      wantedBy = [ "multi-user.target" ];
+      before = [ "i8kmon.service" ];
+      serviceConfig = {
+        Type = "oneshot";
+        RemainAfterExit = true;
+        ExecStart = "${lib.getExe cfg.package} 0";
+        ExecStop = "${lib.getExe cfg.package} 1";
+        Restart = "no";
+      };
+    };
+
+    # see ref in aur: https://aur.archlinux.org/cgit/aur.git/tree/dell-bios-fan-control-resume.service?h=dell-bios-fan-control-git
+    systemd.services.dell-bios-fan-control-resume = {
+      description = "Restart dell-bios-fan-control on resume.";
+      wants = [ "dell-bios-fan-control.service" ];
+      wantedBy = [ "suspend.target" ];
```

Comment:
I don't think I understand what's going on here. The description claims that `dell-bios-fan-control.service` gets restarted when the system resumes operation. Is the idea that whenever the system suspends, it triggers a sleep and then suspends so that when it wakes up, the service will be restarted? Do you happen to know why the service even needs a restart?

No worries if you're not sure and just copying what you're seeing in AUR, I'm just curious :)

Reply:
I suppose this is the answer: https://aur.archlinux.org/packages/dell-bios-fan-control-git#comment-939247

It would be kinda nice to find out what causes the state to reset and make a more dependable target, but I guess that is out of scope of this PR. Maybe you could leave a comment pointing to this discussion though, so anyone reading the code wouldn't have to dig it up?